### ISSUE TITLE
Unable to run dicedb server

### ISSUE THREAD

**Issue by vpsinghg** [DiceDB/dice#885]
OS Specification:

Operating System: linux
Architecture: amd64
CPU Cores: 8
go version go1.23.0 linux/amd64

Steps I followed:

1. Install go 1.23.0

```
$ git clone https://github.com/dicedb/dice
$ cd dice
$ go run main.go

```
**Install GoLangCI**
```
$ sudo su
$ curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /bin v1.60.1

```

**Local Setup with Custom Config**

```
vikram@Vikram-LT:~/dev/real_engineering/dice$ go run main.go
fatal error: runtime: out of memory
runtime stack:
/home/vikram/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.0.linux-amd64/src/runtime/panic.go:1067 +0x48 fp=0x7fbcab7fdc10 sp=0x7fbcab7fdbe0 pc=0x46d3c8

```

I am unable to find any thing unusual in the `htop` command.

Some memory related specifications
```
Total Memory: 7806 MB
Available Memory: 6372 MB
Used Memory: 1126 MB
Free Memory: 3103 MB
Memory Usage: 14.43%
```

**Comment by arpitbbhayani** [comment]
@vpsinghg Can you try changing the config https://github.com/DiceDB/dice/blob/d999d2d6d7d40e709176b18429a54df56c715462/config/config.go#L125 in the source code itself to 10240 and try to run it again? This was a recent change that was pushed and out of memory could have happened because because the internal store impl is reserving more memory during init.

Let me know if this worked out.

**Comment by lucifercr07** [comment]
Can you share your system memory configs and how much free memory is left?
You can try below commands:
`free -h`
`df -h`

**Comment by JyotinderSingh** [comment]
> @lucifercr07 I have added a PR #888 for 1st part. Please review and let me know if there is another approach that you are thinking to avoid this issue.

#888 is now merged.

**Comment by lucifercr07** [comment]
Closing, merged as part of #888 and #908
